pom.xml

```diff
@@ -208,6 +218,11 @@
                     <groupId>org.codehaus.mojo</groupId>
                     <version>3.0.0</version>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-shade-plugin</artifactId>
+                    <version>3.5.1</version>
+                </plugin>
```

Comment:
Do we have strong reasons to upgrade asm and maven-shade-plugin?

Reply:
Yes, we need to upgrade asm and maven-shade-plugin to adapt to higher version of JDK, otherwise there will be some problem in building and testing.